lib/compiler/test/test_open_graph.cc

```diff
@@ -0,0 +1,105 @@
+#include "compiler/unity_algorithm.h"
+#include "doctest.h"
+
+using namespace FlexFlow;
+
+TEST_CASE("get_source_sink_open_graph:basic") {
```

Comment:
Some of these tests may or may not be easier to write using `rapidcheck`--it's worth looking into